packages/core/src/link-refs.ts

```diff
@@ -0,0 +1,38 @@
+import { SolcBytecode } from './solc-api';
+
+export interface LinkReference {
+  src: string;
+  name: string;
+  length: number;
+  start: number;
+  placeHolder: string;
+}
+
+export function extractLinkReferences(bytecode: SolcBytecode): LinkReference[] {
+  const linkRefs: LinkReference[] = [];
+  const { linkReferences } = bytecode;
+  for (const source of Object.keys(linkReferences)) {
+    for (const name of Object.keys(linkReferences[source])) {
+      const { length, start } = linkReferences[source][name][0];
+      const placeHolder = bytecode.object.substr(start * 2, length * 2);
+      linkRefs.push({
+        src: source,
+        name,
+        length,
+        start,
+        placeHolder,
+      });
+    }
+  }
+  return linkRefs;
+}
+
+export function replaceLinkReferences(bytecode: string, linkReferences: LinkReference[]): string {
```

Comment:
I feel like this should be called "unlink bytecode". What do you think?

Reply:
Agree.. actually works much better when this is moved to `version.ts` as it is very related about handling bytecode